File: Day6/AdventDay6_part2.py
import numpy as np

if __name__ == '__main__':
    with open("input6.txt", "r") as f:
        lines = f.readlines()

    # Assuming the first line contains time values and the second line contains distance values
    times = lines[0].split()
    distances = lines[1].split()

    # Remove first element from both lists 
    times.pop(0)
    distances.pop(0)

    # Concatenate the times and distances into single strings
    time = ''.join(times)
    distance = ''.join(distances)

    # Convert the concatenated strings into integers
    time = int(time)
    distance = int(distance)

    ways_you_can_beat = 0

    # Loop over possible values of i from 0 to time (inclusive)
    for i in range(time + 1):
        # Check if the product of i and (time - i) is greater than distance
        if i * (time - i) > distance:
            ways_you_can_beat += 1
    
    # Print the number of ways you can beat
    print(ways_you_can_beat)
    

File: Day6/AdventDay6_part2.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include <sstream>

using namespace std;

int main() {
    ifstream file("input6.txt");  // Open the input file
    if (!file.is_open()) {
        cerr << "Error opening the file!" << endl;
        return 1;
    }

    string line;
    int final_result = 0;  

    // Read all lines from the file into a vector
    vector<string> lines;
    while (getline(file, line)) {
        lines.push_back(line);
    }

    // Ensure there are at least two lines (time and distance)
    if (lines.size() < 2) {
        cerr << "Insufficient data in input file!" << endl;
        return 1;
    }

    // Read the first line as time and the second as distance
    stringstream time_stream(lines[0]);
    stringstream distance_stream(lines[1]);

    string time_value, distance_value, time, distance;

    // Process each pair of time and distance values
    while (time_stream >> time_value && distance_stream >> distance_value) {

        if(isdigit(time_value[0]) && isdigit(distance_value[0])) {
            // Concatenate the time and distance
            time += time_value;
            distance += distance_value;
        }

    }

    // Calculate the number of ways you can beat the condition
    for (long i = 0; i < stoul(time)+1; ++i) {
        if (i * (stoul(time) - i) > stoul(distance)) {
            final_result++;
        }
    }
    // Output the final product
    cout << final_result << endl;

    return 0;
}
